src/plugins/messageStorage/text.js

```diff
@@ -46,24 +46,67 @@ class TextFileMessageStorage {
 		const time = moment(msg.time).utcOffset(tz).format(format);
 		let line = `[${time}] `;
 
-		if (msg.type === "message") {
-			// Format:
+		switch (msg.type) {
+		case "message":
 			// [2014-01-01 00:00:00] <Arnold> Put that cookie down.. Now!!
 			line += `<${msg.from.nick}> ${msg.text}`;
-		} else {
-			// Format:
-			// [2014-01-01 00:00:00] * Arnold quit
-			line += `* ${msg.from.nick} `;
+			break;
+		case "action":
+			// [2014-01-01 00:00:00] * Arnold is eating cookies
+			line += `* ${msg.from.nick} ${msg.text}`;
+			break;
+		case "notice":
+			// [2014-01-01 00:00:00] -Arnold- pssst... I have cookies!
+			line += `-${msg.from.nick}- ${msg.text}`;
+			break;
+		case "join":
+			// [2014-01-01 00:00:00] *** Joins: Arnold (~arnold@foo.bar)
+			line += `*** Joins: ${msg.from.nick} (${msg.hostmask})`;
+			break;
+		case "part":
+			// [2014-01-01 00:00:00] *** Parts: Arnold (~arnold@foo.bar) (Bye all!)
+			line += `*** Parts: ${msg.from.nick} (${msg.hostmask}) (${msg.text})`;
+			break;
+		case "quit":
+			// [2014-01-01 00:00:00] *** Quits: Arnold (~arnold@foo.bar) (Connection reset by peer)
+			line += `*** Quits: ${msg.from.nick} (${msg.hostmask}) (${msg.text})`;
```

Comment:
Shouldn't joins, parts, and quits be something like "foo has quit" instead of "Quits: foo" to match nick, kick, mode, etc.?

Reply:
@xPaw's goal here was to match an existing, common log format, and that's how znc is doing it.

Reply:
I get that, but I think consistency would make more sense than following ZNC's format blindly. We're not bound to ZNC in any way. Thoughts?

Reply:
The idea was to follow an established log format (not ZNC specifically, as theirs is based upon energymech).

https://pisg.github.io/ has a nice list of different log formats, if there's a better looking one, we could switch to that.